sympy/polys/numberfields/subfield.py

```diff
@@ -93,40 +93,45 @@ def field_isomorphism_pslq(a, b):
         A = a.root.evalf(n)
         B = b.root.evalf(n)
 
-        basis = [1, B] + [ B**i for i in range(2, m) ] + [A]
+        basis = [1, B] + [ B**i for i in range(2, m) ] + [-A]
 
-        dps, mp.dps = mp.dps, n
-        coeffs = pslq(basis, maxcoeff=int(1e10), maxsteps=1000)
-        mp.dps = dps
+        dps = mp.dps
+        try:
+            mp.dps = n
+            coeffs = pslq(basis, maxcoeff=int(1e10), maxsteps=1000)
+        finally:
+            mp.dps = dps
```

Comment:
It would be better to use a local context:
```python
In [30]: s2 = sqrt(2).evalf(200)

In [31]: from mpmath import mp

In [32]: ctx = mp.clone()

In [33]: ctx.dps = 100

In [34]: ctx.pslq([1, s2._mpf_, (s2**2)._mpf_])
Out[34]: [2, 0, -1]
```

Reply:
Could we use `workdps`? As in,

```python
from mpmath import pslq, workdps
with workdps(n):
    coeffs = pslq(basis)
```

Reply:
It seems to me that `workdps` will do the same `try - finally` operation under the hood. So it could be used. Maybe it would even be preferable.

Reply:
Does `workdps` modify the `dps` for the global `mp` context? Is it thread-safe if another thread is using the `mp` context as well?

Reply:
`workdps` is defined in `mpmath/__init__.py` as `mp.workdps` where `mp = MPContext()`. I guess we could use ` with MPContext().workdps(n):` to get a temporary local context.

Reply:
Yes, that's better than using a clone of `mp` because a user might have altered the properties of the global `mp` context. I'm not sure we can use with and then use the global `mpmath.pslq` function though because I think that still gets its properties from the global context somehow. So I think it's:
```python
ctx = MPContext()
ctx.dps = n
coeffs = ctx.pslq(basis)
```
We shouldn't need to use `with` since this only modifies the properties of the throwaway `ctx`.

Reply:
> We shouldn't need to use with

Agreed. I also came to that conclusion.